src/model.ts

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as common from '@google-cloud/common';
+import {Dataset} from './dataset';
+
+/**
+ * Model objects are returned by methods such as {@link Dataset#model} and
+ * {@link Dataset#getModels}.
+ *
+ * @class
+ * @param {Dataset} dataset {@link Dataset} instance.
+ * @param {string} id The ID of the model.
+ *
+ * @example
+ * const {BigQuery} = require('@google-cloud/bigquery');
+ * const bigquery = new BigQuery();
+ * const dataset = bigquery.dataset('my-dataset');
+ *
+ * const model = dataset.model('my-model');
+ */
+class Model extends common.ServiceObject {
+  constructor(dataset: Dataset, id: string) {
+    const methods = {
+      /**
+       * Delete a model.
```

Comment:
Maybe "Delete this model." since we're in the context of a specific model.

Reply:
Going to follow suit with the other methods and use "Delete the model", implies the same thing I think.

Reply:
SGTM!